src/crux/lru.clj

```diff
@@ -26,23 +27,20 @@
         lock (StampedLock.)]
     (reify
       LRUCache
-      (compute-if-absent [_ k f]
-        (if (.containsKey cache k)
-          (let [stamp (.writeLock lock)]
-            (try
-              (.computeIfAbsent cache k (reify Function
-                                          (apply [_ k]
-                                            (f k))))
-              (finally
-                (.unlock lock stamp))))
-          (let [v (f k)
-                stamp (.writeLock lock)]
-            (try
-              (.computeIfAbsent cache k (reify Function
-                                          (apply [_ k]
-                                            v)))
-              (finally
-                (.unlock lock stamp))))))
+      (compute-if-absent [_ k stored-key-fn f]
+        (let [v (.getOrDefault cache k ::not-found)] ; use ::not-found as values can be falsy
```

Comment:
Actually, this read also needs to use a lock I realised, sorry. Easiest way is to just delegate to `(get this k ::not-found)` which already does this, see the implementation of `ILookup` a few lines below. Calling `.valAt` explicitly (instead of `get`) would likely be faster.